setup.py

```diff
@@ -33,7 +33,7 @@
         "azure-core>=1.7.0",
         "azure-datalake-store>=0.0.46,<0.1",
         "azure-identity",
-        "azure-storage-blob>=12.5.0",
+        "azure-storage-blob>=12.12.0",
```

Comment:
Blob versioning related adlfs changes require using the current azure service version and pagination changes from 12.12.0:

https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/storage/azure-storage-blob/CHANGELOG.md#12120b1-2022-04-14